nx-X11/programs/Xserver/hw/nxagent/NXdixfonts.c

```diff
@@ -999,1393 +622,86 @@ doListFontsAndAliases(ClientPtr client, LFclosurePtr c)
-		{
-		    UseFPE(fpe);/* since it'll be decref'd later when freed
-				 * from the old list */
-		}
-		else
-		    fpe = 0;
-	    }
-	    /* if error or can't do it, act like it's a new one */
-	    if (!fpe)
-	    {
-		fpe = (FontPathElementPtr) xalloc(sizeof(FontPathElementRec));
-		if (!fpe) 
-		{
-		    err = BadAlloc;
-		    goto bail;
-		}
-		fpe->name = (char *) xalloc(len + 1);
-		if (!fpe->name) 
-		{
-		    xfree(fpe);
-		    err = BadAlloc;
-		    goto bail;
-		}
-		fpe->refcount = 1;
-    
-		strncpy(fpe->name, (char *) cp, (int) len);
-		fpe->name[len] = '\0';
-		fpe->name_length = len;
-		fpe->type = DetermineFPEType(fpe->name);
-		if (fpe->type == -1)
-		    err = BadValue;
-		else
-		    err = (*fpe_functions[fpe->type].init_fpe) (fpe);
-		if (err != Successful)
-		{
-                    #ifndef NXAGENT_SERVER
-		    if (persist)
-		    {
-			ErrorF("Could not init font path element %s, removing from list!\n",
-			       fpe->name);
-		    }
-                    #endif
-		    xfree (fpe->name);
-		    xfree (fpe);
-		}
-	    }
-	}
-	if (err != Successful)
-	{
-	    if (!persist)
-		goto bail;
-	}
-	else
-	{
-	    fplist[valid_paths++] = fpe;
-	}
-	cp += len;
-    }
-
-    FreeFontPath(font_path_elements, num_fpes, FALSE);
-    font_path_elements = fplist;
-    if (patternCache)
-	EmptyFontPatternCache(patternCache);
-    num_fpes = valid_paths;
-
-    return Success;
-bail:
-    *bad = i;
-    while (--valid_paths >= 0)
-	FreeFPE(fplist[valid_paths]);
-    xfree(fplist);
-    return FontToXError(err);
-}
-
-/* XXX -- do we need to pass error down to each renderer? */
-int
-SetFontPath(ClientPtr client, int npaths, unsigned char *paths, int *error)
-{
-    int   err = Success;
-
-    if (npaths == 0) {
-	if (SetDefaultFontPath(defaultFontPath) != Success)
-	    return BadValue;
-    } else {
-	err = SetFontPathElements(npaths, paths, error, FALSE);
-    }
-    return err;
-}
-
-int
-SetDefaultFontPath(char *path)
-{
-    unsigned char *cp,
-               *pp,
-               *nump,
-               *newpath;
-    int         num = 1,
-                len,
-                err,
-                size = 0,
-                bad;
-
-    /* get enough for string, plus values -- use up commas */
-#ifdef NX_TRANS_SOCKET
-    len = strlen(_NXGetFontPath(path)) + 1;
-#else
-    len = strlen(path) + 1;
-#endif
-    nump = cp = newpath = (unsigned char *) ALLOCATE_LOCAL(len);
-    if (!newpath)
-	return BadAlloc;
-#ifdef NX_TRANS_SOCKET
-    pp = (unsigned char *) _NXGetFontPath(path);
-#else
-    pp = (unsigned char *) path;
-#endif
-    cp++;
-    while (*pp) {
-	if (*pp == ',') {
-	    *nump = (unsigned char) size;
-	    nump = cp++;
-	    pp++;
-	    num++;
-	    size = 0;
-	} else {
-	    *cp++ = *pp++;
-	    size++;
-	}
-    }
-    *nump = (unsigned char) size;
-
-    err = SetFontPathElements(num, newpath, &bad, TRUE);
-
-    DEALLOCATE_LOCAL(newpath);
-
-    return err;
-}
-
-unsigned char *
-GetFontPath(int *count, int *length)
```

Comment:
Good catch.

Actually, we change dix/dixfonts.c here in commit aea71067dccd63e5988a57944991a2775ae4e5a7 without duplicating that change to NXdixfonts.c.

This was before we realized these massive code duplications in NX*.c files.